tools/autograd/load_derivatives.py

```diff
@@ -606,3 +616,25 @@ def dedup_vars(vars: Sequence[SavedAttribute]) -> Sequence[SavedAttribute]:
         seen.add(name)
         saved.append(var)
     return saved
+
+
+K = typing.TypeVar('K')
+V = typing.TypeVar('V')
+
+def merge_dicts(*dicts: Dict[K, V]) -> Dict[K, V]:
+    """Merges dictionaries, raising in the event of a conflict."""
+    out: Dict[K, V] = {}
+    for d in dicts:
+        for k, v in d.items():
+            insert_or_raise(out, k, v)
+    return out
+
+
+def insert_or_raise(d: Dict[K, V], key: K, value: V) -> None:
```

Comment:
Do we need to worry about duplicates? Or wouldn't you only end up with duplicates if there were multiple outputs to an operator with the same name - which would have already been an error further upstream anyway.

Reply:
Duplicates should not be able to happen as the code is currently written. If named outputs are unique, then there is a 1:1 correspondence between each name and its array index, and this mapping flows through to the codegen.

Nevertheless, I would like to at least assert that we do not mess that up.

Reply:
I think this is resolved now with the changes from the other comment.